core/unit_test/TestCrs.hpp

```diff
@@ -63,11 +63,35 @@ struct CountFillFunctor {
   }
 };
 
+template< class CrsType, class ExecSpace, class scalarType >
+struct RunUpdateCrsTest {
+
+  CrsType graph {};
+  RunUpdateCrsTest( CrsType g_in ) : graph(g_in)
+  {
+  }
+
+  void run_test() {
+     parallel_for ("TestCrs1", Kokkos::RangePolicy<ExecSpace>(0,graph.numRows()),*this);
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const scalarType row) const {
+     auto row_map = graph.row_map;
+     auto entries = graph.entries;
+     auto j_start = (row >= static_cast<decltype(row)>(row_map.extent(0))) ? 0 : row_map(row);
```

Comment:
We don't need the safety checks here since the row iteration count only goes to number of rows which is smaller than row_map.extent(0). Also the parallel for starts at 0. As a general rule we only do runtime checks where they are necessary. The parallel_for tests itself need to ensure that the contract of what reaches the operator doesn't get violated.